// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

/*option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";*/

package modelresponse;

// The greeting service definition.
service ModelResponse {
  // Sends a greeting
  rpc GeneratorReply (SingleStringRequest) returns (SingleStringReply) {}
  rpc ClassificationReply (SingleStringRequest) returns (SingleStringReply) {}
  rpc QuestionAndAnswerReply(QARequest) returns (SingleStringReply) {}
  rpc FillMaskReply(SingleStringRequest) returns (SingleStringReply) {}
  rpc TokenClassificationReply(SingleStringRequest) returns (SingleStringReply) {}
  rpc ConversationalReply(ConversationRequest) returns (ConversationReply) {}
}

message Value {
    oneof oneof_values {
        string svalue = 1;
        int64 ivalue = 2;
        float fvalue = 3;
        bool bvalue = 4;
    }
}

message SingleStringRequest {
  repeated string request = 1;
  map<string,Value> query_kwargs = 2;
}

message SingleStringReply {
  repeated string response = 1;
  float time_taken = 2;
}

message QARequest {
  string question = 1;
  string context = 2;
  map<string,Value> query_kwargs = 3;
}

message ConversationRequest {
  string text = 1;
  optional int64 conversation_id = 2;
  repeated string past_user_inputs = 3;
  repeated string generated_responses = 4;
  map<string,Value> query_kwargs = 5;
}

message ConversationReply {
  int64 conversation_id = 1;
  repeated string past_user_inputs = 2;
  repeated string generated_responses = 3;
  float time_taken = 4;
}
